models/user/user.go

```diff
@@ -261,38 +259,38 @@ func (u *UserStruct) Projects() []*project.ProjectStruct {
 	return slice
 }
 
-func (u *UserStruct) Save() bool {
+func (u *UserStruct) Save() error {
 	table := u.database.Init()
 	defer table.Close()
 
 	result, err := table.Exec("insert into users (email, password, provider, oauth_token, uuid, user_name, avatar_url, created_at) values (?, ?, ?, ?, ?, ?, ?, now());", u.Email, u.Password, u.Provider, u.OauthToken, u.Uuid, u.UserName, u.Avatar)
 	if err != nil {
 		logging.SharedInstance().MethodInfo("user", "Save", true).Errorf("failed to create user: %v", err)
-		return false
+		return err
 	}
 	u.ID, _ = result.LastInsertId()
 	logging.SharedInstance().MethodInfo("user", "Save").Infof("user saved: %v", u.ID)
-	return true
+	return nil
 }
 
-func (u *UserStruct) Update() bool {
+func (u *UserStruct) Update() error {
 	table := u.database.Init()
 	defer table.Close()
 
 	_, err := table.Exec("update users set provider = ?, oauth_token = ?, uuid = ?, user_name = ?, avatar_url = ? where email = ?;", u.Provider, u.OauthToken, u.Uuid, u.UserName, u.Avatar, u.Email)
 	if err != nil {
 		logging.SharedInstance().MethodInfo("User", "Update", true).Panic(err)
-		return false
+		return err
 	}
-	return true
+	return nil
 }
 
-func (u *UserStruct) CreateGithubUser(token string, githubUser *github.User, primaryEmail string) bool {
+func (u *UserStruct) CreateGithubUser(token string, githubUser *github.User, primaryEmail string) error {
```

Comment:
__[golint]__
* [golint] exported method UserStruct.CreateGithubUser should have comment or be unexported